### PR TITLE
ST: upgrade tests improve

### PR DESCRIPTION
**Pull request by Frawless**
### Type of change

- Enhancement / new feature

### Description

Improve upgrade tests and add 0.13.0 version to the matrix. Partly address https://github.com/strimzi/strimzi-kafka-operator/issues/1791, but still we need more refactor which @Seequick1  is working on currently.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

File: systemtest/pom.xml [hunk 1]
```diff
@@ -98,6 +98,25 @@
             <version>${jupiter.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>net.joshka</groupId>
+            <artifactId>junit-json-params</artifactId>
+            <version>5.5.1-r0</version>
+        </dependency>
+
```
Comment (scholzj) [inline systemtest/pom.xml]: Can you remove the empty lines here?

Comment (Frawless) [reply systemtest/pom.xml]: Done

File: systemtest/src/main/java/io/strimzi/systemtest/Environment.java [hunk 2]
```diff
@@ -54,6 +54,10 @@
      * CO reconciliation interval.
      */
     private static final String STRIMZI_FULL_RECONCILIATION_INTERVAL_MS_ENV = "STRIMZI_FULL_RECONCILIATION_INTERVAL_MS";
+    /**
+     * Openshift 4 parameter.
+     */
+    private static final String OPENSHIFT4_ENV = "OPENSHIFT4";
```
Comment (tombentley) [inline systemtest/src/main/java/io/strimzi/systemtest/Environment.java]: Why does the correct functioning of the strimzi upgrade depend on the OpenShift4-ness of the cluster?

Comment (Frawless) [reply systemtest/src/main/java/io/strimzi/systemtest/Environment.java]: Upgrade from `0.8.2` to `0.11.4` is not working on ocp4 and I didn't find better solution how to determine which cluster version we are using in the tests.

Comment (tombentley) [reply systemtest/src/main/java/io/strimzi/systemtest/Environment.java]: Is there an issue about that? I don't see one. We should look into understanding why it doesn't work on that platform, and only add a _temporary_ workaround like this if and when we know that it's not a simple fix.

Comment (scholzj) [reply systemtest/src/main/java/io/strimzi/systemtest/Environment.java]: Realistically, 0.8.2 existed long before OpenShift 4. And I do not think we will do any reelases for such an old version to get the upgrades working. I guess the only interesting question is whether it is OpenShift 4+ or Kubernetes 1.12+.

**Pull request merged**